app/assets/javascripts/active_admin/contributions.js

```diff
@@ -0,0 +1,22 @@
+$(window).on("load", function () {
+  $("a[name=refuse_contribution]").on("click", function () {
+    const reasons = Object.values($(this).data("reason"));
+    const id = $(this).data("id");
+    ActiveAdmin.ModalDialog(
+      "Justificativa:",
+      { Motivo: reasons },
+      function (inputs) {
+        $.ajax({
+          url: `/admin/contributions/${id}/refuse`,
+          data: JSON.stringify({ rejected_reason: inputs.Motivo }),
+          contentType: "application/json; charset=utf-8",
+          dataType: "json",
+          type: "PUT",
+          complete: function (result) {
+            window.location.reload();
+          },
```

Comment:
Hey, @HeitorMC. Screw this comment. As you're redirecting after calling the `refuse` action it should show up properly message when something goes wrong as well.

It looks good to me!

Reply:
You can mark it as resolved (I'm no allowed to do that)

Reply:
please check if it's really the case

Reply:
Hmmm, I don't think it's the case because the `refuse` will only appear if the contribution state is `received`. And still, jQuery will request to rails itself, without user params. All the parameters are passed automatically through active admin. So, the only faulty case that I foresee, would be if this route `/admin/contributions/100/refuse` stopped working.

What do you guys think?

Reply:
Actually, this comment was solved here ce35bb447c077a847e76ccd0b5851fef658d3ec8. So right now we do have a transaction to execute those updates and if something fails, there is a rescue clause for it. Assuming that the action fails, a redirect will occur and an alert will be shown.